ext/ManoptJuMPExt.jl

```diff
@@ -34,6 +34,20 @@ struct RiemannianFunction{MO<:Manopt.AbstractManifoldObjective} <:
     func::MO
 end
 
+MOI.Utilities.map_indices(::Function, func::RiemannianFunction) = func
+
+# We we don't support `MOI.modify` and `RiemannianFunction` is not mutable, no need to copy anything
+Base.copy(func::RiemannianFunction) = func
+
+# This is called for instance when the user does `@objective(model, Min, func)`.
```

Comment:
Thanks for this thorough comment in the code. Very helpful to learn what this wrapping is good for :)